go/libraries/doltcore/sqle/dfunctions/reset.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dfunctions
+
+import (
+	"github.com/dolthub/dolt/go/libraries/doltcore/sqle"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+const ResetHardFuncName = "reset_hard"
+
+type ResetHardFunc struct{}
+
+// NewResetHardFunc creates a new ResetHardFunc expression.
+func NewResetHardFunc() sql.Expression {
```

Comment:
See my top level comment, but I think for these it's important to exactly mirror the semantics of the command line. So this should be `dolt_reset` and take N args: a boolean `hard` param followed by N table names.

Reply:
(hard == false would be a no-op)